topology/probes/istio/virtualservice.go

```diff
@@ -135,6 +135,51 @@ func virtualServicePodAreLinked(a, b interface{}) bool {
 	return false
 }
 
+func findMissingGateways(g *graph.Graph) {
```

Comment:
this implementation yeilds complexity of `O(n^2)` per each virtualservice/gateway node event:

- loop through all virtualservices (line 145) ~ O(n)
- loop through all gateways (line 151) ~ O(n)

This can (and should be) reduced to O(n):
- if virtualservice event --> scan gateways and vis-versa ~ O(n)

Reply:
OK